File: src/0693.binary-number-with-alternating-bits.693/solution.ts
function hasAlternatingBits (n: number): boolean {
    let last = n & 1;
    n >>>= 1;
    while (n > 0) {
        const digit = n & 1;
        if ((digit ^ last) === 0) {
            return false;
        }
        last = digit;
        n >>>= 1;
    }
    return true;
}

File: src/0693.binary-number-with-alternating-bits.693/solution.java
class Solution {
    public boolean hasAlternatingBits(int n) {
        int last = n&1;
        n >>= 1;
        while (n>0){
            int digit = n&1;
            if(digit == last){
                return false;
            }
            last = digit;
            n>>=1;
        }
        return true;
    }
}